### PR TITLE
feat: add js-insights package

### PR DESCRIPTION
**Pull request by jbalsas**
From https://github.com/liferay/liferay-npm-tools/pull/269

Found some time to work on this on the plane, so I'm just dropping it here to see what we want to do with this. 

It's basically a tool that:
- **Given** a set of paths (glob)
- **Then** analyzes the code to gain insights (currently only checking external imports)
- **Then** generates a report

Different reports can be generated:

### Table (default)
report_table

### Airtable
See real [liferay-portal report](https://airtable.com/tbln9P08dd75xKAmJ/viw2InOW91BooKGuV)

report_airtable

### JSON
report_json

As I was building it, I kept thinking that we could easily add other insights like "`setInterval` usage" throughout our base code that could help us make better informed decisions.

The code is very rough... specially in the `reporters` area where I haven't really thought how to approach it. For now, reports just know what to expect from the only insight we have right now.

Anyways, creating this as a draft to see where this takes us...

### CODE REVIEW

File: packages/liferay-js-insights/reporters/table.js [hunk 1]
```diff
@@ -6,23 +6,36 @@
...
 const Table = require('cli-table');
 
+const DEFAULT_CONFIG = {
+	output: 'app,name,dependencies.clay3,dependencies.react'
+};
+
+/**
+ * Traverses an object provided a path to a property and returns its value if found or null if the path can't be reached.
+ */
+const idx = (p, o) => p.reduce((xs, x) => (xs && xs[x] ? xs[x] : null), o);
```
Comment (wincent) [inline packages/liferay-js-insights/reporters/table.js]: Bet you can't wait for https://github.com/tc39/proposal-optional-chaining 😂

File: packages/liferay-js-insights/.yarnrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Make `yarn version` produce the right commit message and tag for this package.
+version-tag-prefix "liferay-js-insights/v"
+version-git-message "chore: prepare lifeiray-js-insights/v%s"
```
Comment (wincent) [inline packages/liferay-js-insights/.yarnrc]: Please fix `lifeiray` to `liferay` before merging.

Comment (jbalsas) [reply packages/liferay-js-insights/.yarnrc]: 🙈 I swear I did fix this one... probably lost the commit 🤔

File: packages/liferay-js-insights/reporters/table.js (hunk 1 above)
Comment (jbalsas) [reply packages/liferay-js-insights/reporters/table.js]: Good things worth waiting for ;)

**Comment by jbalsas** [comment]
Typo fixed!

**Pull request merged**